addon/components/base/bs-form.js

```diff
@@ -136,6 +137,31 @@ export default Component.extend({
    */
   horizontalLabelGridClass: 'col-md-4',
 
+  /**
+   * `isSubmitting` is `true` after `submit` event has been triggered and until Promise returned by `onSubmit` is
+   * fulfilled. If `validate` returns a Promise that one is also taken into consideration.
+   *
+   * `isSubmitting` property is yielded and could be used to customize form submit button. E.g. a loading spinner could
+   * be shown and the submit button been disabled:
+   *
+   * ```
+   * {{#bs-form onSubmit=(action 'save') as |form|}}
+   *   {{#bs-button type='submit' disabled=form.isSubmitting}}
+   *     Save {{fa-icon 'spinner'}}
+   *   {{/bs-button}}
+   * {{/bs-from}}
```

Comment:
Thinking further, does it maybe even make sense to align the APIs here, so we yield something like `submitState` or so, which has all the states introduced in the button component, like `isPending` (equivalent to `isSubmitting`), but also `isResolved` and `isRejected`?

Reply:
I'm definitely in favor with a `{{#form.button}}` component but I think it's out of scope for this PR. Should also address #92.

I don't like yielded hash. It's quite complex API to use. You would end up with lot of code, which is hard to read:

```
{{#bs-button disabled=submitState.isPending as |button|}}
  {{#if submitState.isResolved}} Saved {{else}} Save {{/if}}
  {{#if submitState.isPending}} {{fa-icon 'spinner'}} {{/if}}
{{/bs-button}}
```

IMO `isSubmitting` and `isSubmitted` are easier to understand. At least the first one would also be inline with `isSaving` property of `DS.Model`.
```
{{#bs-button disabled=isSubmitting as |button|}}
  {{#if isSubmitted}} Saved {{else}} Save {{/if}}
  {{#if isSubmitting}} {{fa-icon 'spinner'}} {{/if}}
{{/bs-button}}
```

Not quite sure about an `isRejected` equivalent. Not only because of naming but also cause it does not to distinguish between failed validation, server-side validation error and other server-side errors. So might just lead to wrong assumptions.

It's a different story for a `{{bs-button}}` cause we don't know what the attached action is about.